### ISSUE TITLE
Some apps show backup errors

### ISSUE THREAD

**Issue by grote** [stevesoltys/seedvault#147]
In the app status screen, some apps have been reported to always produce errors (red circle icon).

## K/V Backup without any data

These apps claim to support backups (old style key-value), but when the system tries to back them up, their backup agent is not giving us any data.

* WhatsApp
* Twidere

## To investigate

* Waze
* Google Maps
* Google Pay

**Comment by chirayudesai** [comment]
* Google Maps is an old issue, it has a backup agent set that's in Play Services which we miss and thus the failure.
* Waze _might_ be similar
* Google Pay being a payment app, could be anything.

**Comment by ForceGT** [comment]
I think WhatsApp doesn't use the `allowBackup` flag which actually allows Android's native implementation for backup to go in and grab all the data from `/data/` and convert it to the `.ab` file. I mean atleast `adb backup` works that way. And there are many projects which as a workaround install the older version of WhatsApp which has the `allowBackup` set to true in the manifest which technically does allow backups and returns some data rather than blank. Just thought I should add in my two cents on the matter here

**Comment by vbooka1** [comment]
Hello, I have a red circle with `!` exclamation mark for one app that weights 8.2 GB. There is no any text message, just the icon.
I have more than enough free space for a local backup. And, of course, the "Unlimited app quota" checkbox is toggled on.
There is only one app with that red "!" circle, and this is the largest app on my phone. Maybe there is some hidden hardcoded limit for a single app like 4 GB maximum?

How do I check the backup logs?

**Comment by grote** [comment]
> How do I check the backup logs?

expert settings where you toggled an "Unlimited app quota".

However, 8.2 GB is a lot to back up. What backup location to you back up to? Something internet based? A FAT32 file-system maybe?

**Comment by vbooka1** [comment]
> What backup location to you back up to? Something internet based? A FAT32 file-system maybe?

No, that's the first thing I've thought about - something like FAT32 limit of 4GB per file, but I make backup to the local storage which should be F2FS or EXT4 or something like that.

```
09-23 22:33:33.345  1551 24206 D BackupManagerService: Calling doFullBackup() on com.divessi.ssi
...
09-23 22:38:33.377  1551  3591 D BackupManagerService: Timeout message received for token=641a14ad
09-23 22:38:33.377  1551  3591 V LifecycleOperationStorage: [UserID:0] Cancel: token=641a14ad
09-23 22:38:33.377  1551  3591 W PFTBT   : Full backup cancel of com.divessi.ssi
```

The time difference between start and timeout error is exactly 5 minutes, is that a hardcoded value?

**Comment by vbooka1** [comment]
Well, of course it is a hardcoded value because it is a "timeout".

> com.divessi.ssi

It is an application with books about diving (SCUBA, freediving, etc) and marine ecology, where each "book" is actually an offline website consisting of hundreds of separate files - html, css, images, movies, et cetera. If you would download multiple books you will have up to several thousands of files, so I believe the underlying problem is not the total size of the application but the amount of files: it takes just too long to create an archive of a directory with thousands of files inside.

Please consider making a larger timeout, 10 minutes maybe?

**Comment by grote** [comment]
> The time difference between start and timeout error is exactly 5 minutes, is that a hardcoded value?

Yes, this is what the system gives us to finish a backup. What ROM do you use?

ROMs should have the option to increase those timeouts. I wrote about this here: https://github.com/seedvault-app/seedvault/issues/564#issuecomment-1753285697

You may be able to bump them with `adb`.

**Comment by vbooka1** [comment]
> What ROM do you use?

GrapheneOS Android 13

> ROMs should have the option to increase those timeouts. I wrote about this here: https://github.com/seedvault-app/seedvault/issues/564#issuecomment-1753285697

I'm afraid there is no such option in GrapheneOS flavor of Android. 
I've checked GrapheneOS Android 14 on another phone and did not find it either, the only difference from Android 13 is a new option in the expert settings, "Device-to-device backup".

**Comment by grote** [comment]
You then need to take it up with your ROM to increase those timeouts or the app developer to only include essential files in the backup (nothing that can be downloaded again).

**Comment by vbooka1** [comment]
The whole point of making a backup is to not to download 8 gigabytes again...

**Comment by baack** [comment]
Why should app be allowed to decide what to backup and what to refuse? Why not user? 

Isn't is possible to force-backup all app's data no matter what this app desire and says?
